rust/tools/authorization-logic/src/parsing/AuthLogic.g4

```diff
@@ -81,8 +81,19 @@ importAssertion
     : IMPORT principal SAYS ID
     ;
 
+// The name "type" would cause name collisions in the code that antlr generates
+authLogicType
+    : NUMBERTYPE #numberType
+    | PRINCIPALTYPE #principalType
```

Comment:
Any reason to keep this separate from `ID #customType`?

Reply:
Not too many strong reasons at the moment. Some weak reasons though:
 - Extensibility --> extending the grammar of principals is one thing we might naturally do (add and/or on principals for example), and making this a builtin type could help with that. (A bit moot in this case since this is not the implementation we will keep around).
 - Principals can show up in more parts of the grammar than objects of the other types. For example in `canSay` or `canActAs`, so in the implementation when these need to be generated we can call PrincipalType instead of CustomType.

Reply:
I'm assuming that the approval means that this is OK to leave for now? If you'd rather get rid of this LMK. We can also just keep this in mind for the C++ impl.